dolt/middleware.py

```diff
@@ -66,6 +80,12 @@ def process_view(self, request, view_func, view_args, view_kwargs):
     def get_branch(self, request):
         # lookup the active branch in the session cookie
         requested = branch_from_request(request)
+
+        # override if the query string exists
+        queryStringBranch = request.GET.get(DOLT_BRANCH_KEYWORD, None)
```

Comment:
We should separate this logic from this method, we don't want to have to different branch state parameters.
Instead, we should intercept this querystring parameter above line 44, before we check the session cookie. 
If we see the query string parameter, we remove it from the url, update the session cookie, and redirect to our simplified URL